rock-paper-scissors/service/src/main.rs

```diff
@@ -0,0 +1,218 @@
+use oasis_std::{Address, Context};
+#[macro_use]
+extern crate serde;
+
+#[derive(oasis_std::Service)]
+struct RockPaperScissors {
+    player_name: String,
+    player_move: Move, // TODO would be better as an enum?
+    challenger_name: String,
+    challenger_move: Move,
+    played: bool, // whether or not someone can challenge. true = player has played. false = nobody has played or just finished.
+    challenged: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum Move {
+    Null,
+    Rock,
+    Paper,
+    Scissors,
+}
+
+impl Move {
+    fn compare(&self, challenger: &Self) -> i32 {
+        if self.eq(&Move::Null) || challenger.eq(&Move::Null) {
+            panic!("Trying to compare Null move!");
+        }
+        else if self.eq(&Move::Rock) {
+            if challenger.eq(&Move::Rock) {
+                return 0;
+            } else if challenger.eq(&Move::Scissors) {
+                return 1;
+            } else {
+                return -1;
+            }
+        }
+        else if self.eq(&Move::Paper) {
+            if challenger.eq(&Move::Rock) {
+                return 1;
+            } else if challenger.eq(&Move::Scissors) {
+                return -1;
+            } else {
+                return 0;
+            }
+        }
+        else {
+            if challenger.eq(&Move::Rock) {
+                return -1;
+            } else if challenger.eq(&Move::Scissors) {
+                return 0;
+            } else  {
+                return 1;
+            }
+        }
+    }
+
+    fn is_valid(&self) -> bool {
+        if !self.eq(&Move::Rock) && !self.eq(&Move::Paper) && !self.eq(&Move::Scissors) {
```

Comment:
why are you allowed to make invalid moves at all? I don't think that `Move::Null` is necessary. Try `player_move: Option<Move>`